caom2pipe/client_composable.py

```diff
@@ -431,29 +431,25 @@ def look_pull_and_put(
         just the checksum part of ChecksumURI please, or the comparison will
         always fail.
     """
-    retrieve = False
     cadc_meta = cadc_client.info(storage_name)
     if (
         (
             checksum is not None and
             cadc_meta is not None and
-            cadc_meta.md5sum != checksum
+            cadc_meta.md5sum.replace('md5:', '') != checksum
```

Comment:
Will the `md5sum` field never be `None`?

Reply:
If the flow says that if it is `None`, then this will break, then let CADC know, then great.  That comparison just makes assumptions, that's all.  Thanks.

Reply:
Aye, it does make assumptions. Thank you for the reminder that assumptions in code are bad.